free_fleet/src/utilities/utilities.cpp

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (C) 2021 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+#include "utilities.hpp"
+
+namespace free_fleet {
+
+//==============================================================================
+double distance_to_waypoint(
+  const rmf_traffic::agv::Graph::Waypoint& waypoint,
+  const Eigen::Vector2d& coordinates)
+{
+  const Eigen::Vector2d p = waypoint.get_location();
+  return (p - coordinates).norm();
+}
+
+//==============================================================================
+double distance_to_lane(
+  const rmf_traffic::agv::Graph::Lane& lane,
+  const rmf_traffic::agv::Graph& graph,
+  const Eigen::Vector2d& coordinates)
+{
+  const Eigen::Vector2d entry_loc =
+    graph.get_waypoint(lane.entry().waypoint_index()).get_location();
+  const Eigen::Vector2d exit_loc =
+    graph.get_waypoint(lane.exit().waypoint_index()).get_location();
+  const Eigen::Vector3d entry_p = {entry_loc[0], entry_loc[1], 1.0};
+  const Eigen::Vector3d exit_p = {exit_loc[0], exit_loc[1], 1.0};
+
+  Eigen::Vector3d line_coef = entry_p.cross(exit_p);
+  line_coef = line_coef / line_coef[2];
+
+  const double dist =
```

Comment:
I was using this method, https://en.wikipedia.org/wiki/Distance_from_a_point_to_a_line, with homogeneous coordinates.

I changed it over to follow the implementation you linked, to be consistent and just in case I made a mistake, https://github.com/open-rmf/free_fleet/pull/76/commits/f20193cb3cef29d3e7b89223df106ec8e966a02b.